torch/_inductor/codegen/cuda/cuda_template.py

```diff
@@ -69,30 +74,51 @@ def __init__(
     def supports_epilogue_fusion(op: GemmOperation) -> bool:
         return False
 
-    def generate(  # type: ignore[override]
-        self,
-        description,
-        **kwargs,
-    ) -> CUDATemplateCaller:
+    def make_key(self, op: "GemmOperation") -> str:
         """
-        Generates the CUDA template caller object for the given GEMM template and operation. This CUDATemplateCaller
-        may be used to call and benchmark the generated CUDA kernel in a standalone manner to enable Autotuning.
+        Make a key for the code cache. The idea of the method is to cache
+        everything that matters but doesn't include runtime param values, i.e.,
+        self.get_runtime_arg_values().
 
         Args:
-            kwargs: Additional keyword arguments.
-
-        Returns:
-            A CUDATemplateCaller object representing the generated CUDA template caller.
+            kwargs: Additional keyword arguments. Including op (GemmOperation).
+        """
+        return hashlib.sha256(
+            str(
+                (
+                    create_inputs_key(self.input_nodes),
+                    self.input_reorder,
+                    # output layout, same as self.output_node.get_layout()
+                    self.layout,
+                    self.get_runtime_arg_info(),
+                    op.configuration_name(),
+                )
+            ).encode("utf-8")
+        ).hexdigest()
+
+    def generate_code_and_args(self, **kwargs) -> tuple[str, tuple[int, ...]]:
         """
+        Generate code and args with caching. We cache the code even if runtime
+        args are different.
```

Comment:
can add more into 
"We cache the code even if runtime args are different."
which is ok because ....